### PR TITLE
users/ package docstring

### PR DESCRIPTION
**Pull request by boyska**
It tries to describe the whole architecture about how users are handled,
and what this package is not.

### CODE REVIEW

File: users/__init__.py [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+
+This also means that a user has no capability (directly). It just belongs to
+groups, which, in turn, have capabilities.
+
+The rationale behind what a Capability is may seem baroque, but there are
+several advantages to it:
+
+- it is decoupled from  the actual domains used by the UI
+- the regular expression make it possible to create groups that can operate on
+  everything (``.*``).
```
Comment (ael-code) [inline users/__init__.py]: In previous line you've used ``/cars/*/tires`` (simplified expression) then here you switched to ``.*`` ( regular expression )

Comment (boyska) [inline users/__init__.py]: ael wrote:
>In previous line you've used ``/cars/*/tires`` (simplified expression)
>then here you switched to ``.*`` ( regular expression )

fixed (moved everything to regular expression)

Comment (ael-code) [inline users/__init__.py]: > fixed (moved everything to regular expression)

are not the ones that we actually use:
`.*` match the empty string and strings with slashes, we use `[^/]+` instead

Comment (boyska) [inline users/__init__.py]: ok, all moved back to simplified form.

File: users/__init__.py [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+- A :py:class:`~users.models.Group` is a collection of users. Note that a user can belong to multiple
+  groups. A group has capabilities.
+- A :py:class:`~users.models.Capability` is a "granted permission". You can think of it like a piece of
+  paper saying, ie. "you can create new attachments".
+
+  - Its :py:attr:`~users.models.Capability.action` is a composition of Create, Read, Update, Delete
+    (it follows the CRUD model).
+  - A :py:attr:`~users.models.Capability.domain` is a regular expression that
+    must "match" to the description of an object. ie. ``/cars/*`` means "every
+    car", while ``/cars/*/tires`` means "the tires of every car"
```
Comment (ael-code) [inline users/__init__.py]: > while ``/cars/*/tires`` means "the tires of every car"

It should be ``/cars/*/tires/*``

Comment (boyska) [inline users/__init__.py]: On Sat, Nov 21, 2015 at 03:50:27AM -0800, ael wrote:
>> +capabilities. Note that this package does **not** specify permissions for
>> +objects. Actual permissions are handled at the UI level.
>> +
>> +The main concepts are:
>> +
>> +- A :py:class:`~users.models.User` is what you think it is; something that you can login as.
>> +- A :py:class:`~users.models.Group` is a collection of users. Note that a user can belong to multiple
>> +  groups. A group has capabilities.
>> +- A :py:class:`~users.models.Capability` is a "granted permission". You can think of it like a piece of
>> +  paper saying, ie. "you can create new attachments".
>> +
>> +  - Its :py:attr:`~users.models.Capability.action` is a composition of Create, Read, Update, Delete
>> +    (it follows the CRUD model).
>> +  - A :py:attr:`~users.models.Capability.domain` is a regular expression that
>> +    must "match" to the description of an object. ie. ``/cars/*`` means "every
>> +    car", while ``/cars/*/tires`` means "the tires of every car"
>
>> while ``/cars/*/tires`` means "the tires of every car"
>
>It should be ``/cars/*/tires/*``

fixed

**Pull request merged**